flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -1119,27 +1076,15 @@ private void finishCancellation() {
 		}
 	}
 
-	void cachePartitionInfo(PartialInputChannelDeploymentDescriptor partitionInfo) {
-		partialInputChannelDeploymentDescriptors.add(partitionInfo);
+	void cachePartitionInfo(PartitionInfo partitionInfo) {
+		partitionInfos.add(partitionInfo);
 	}
 
-	void sendPartitionInfos() {
-		// check if the ExecutionVertex has already been archived and thus cleared the
-		// partial partition infos queue
-		if (partialInputChannelDeploymentDescriptors != null && !partialInputChannelDeploymentDescriptors.isEmpty()) {
-
-			PartialInputChannelDeploymentDescriptor partialInputChannelDeploymentDescriptor;
-
-			List<PartitionInfo> partitionInfos = new ArrayList<>(partialInputChannelDeploymentDescriptors.size());
-
-			while ((partialInputChannelDeploymentDescriptor = partialInputChannelDeploymentDescriptors.poll()) != null) {
-				partitionInfos.add(
-					new PartitionInfo(
-						partialInputChannelDeploymentDescriptor.getResultId(),
-						partialInputChannelDeploymentDescriptor.createInputChannelDeploymentDescriptor(this)));
-			}
+	private void sendPartitionInfos() {
+		if (!partitionInfos.isEmpty()) {
+			sendUpdatePartitionInfoRpcCall(Lists.newArrayList(partitionInfos));
```

Comment:
Could we avoid using 3rd party dependencies like Guava? it could be just `new ArrayList<>(partitionInfos)`. Is his defensive copy of `partitionInfos` actually needed?

Reply:
Yeah, I think it makes sense to use `new ArrayList<>(partitionInfos)` directly.